### PR TITLE
Add membership field to Channel and REST API serializer

### PR DESCRIPTION
**Pull request by noisecapella**
#### What are the relevant tickets?
Part of #864 

#### What's this PR do?
Adds a `membership_is_managed` field which defaults to true for channels created via REST API.

#### How should this be manually tested?
 - Create a channel in micromasters and verify that there is a `Channel` object in discussions for that channel and `membership_is_managed` is `True`
 - View `/api/v0/channels/` and verify that the field shows up in the results

### CODE REVIEW

File: channels/serializers.py [hunk 1]
```diff
@@ -58,20 +75,34 @@ def get_user_is_moderator(self, channel):
         """
         return bool(channel.user_is_moderator)
 
+    def get_membership_is_managed(self, channel):
+        """
+        Get membership_is_managed from the associated Channel model
+        """
+        try:
+            channel_obj = Channel.objects.get(name=channel.display_name)
```
Comment (mbertrand) [inline channels/serializers.py]: I've got an older channel where name != display_name, so this line raises the DoesNotExist exception and returns `False` for `membership_is_managed` even though the actual value is `True`; is this the intended behavior?

Comment (noisecapella) [reply channels/serializers.py]: As far as I know `name` should always be the same as `display_name`. When you have a moment can you show me the channel in person?

**Comment by noisecapella** [comment]
Just removed the `except` clause

**Comment by noisecapella** [comment]
I think all channels accessible to the end user should be accessible to the API user. A user could produce this error by typing in the URL to access a channel which exists on the reddit instance but was not created by that particular discussions server, but I feel like having the error would be preferable to just returning false on that value. Though maybe we should add this check to our permissions at some point to avoid extra 500 errors

**Pull request merged**